### ISSUE TITLE
[SUGGESTION] When setting a var to itemtype convert to itemstack

### ISSUE THREAD

**Issue by Fusezion** [SkriptLang/Skript#5996]
### Suggestion

When you set a variable to say `any sword` we should at that moment in time convert it to an itemstack and use the `getRandom` method.

### Why?

In skript currently when we set a variable to itemtype it's object is an itemtype this allows for situations where the item is undetermined until somewhere else in skript such as a `give` effect calls `getRandom()` so code like below can happen.
```
set {_item} to any sword
loop players:
    give loop-player {_item}
    # Player 1 gets stone
    # Player 2 gets diamond
    # Player 3 gets wooden
```
This should only happen if we use `give loop-player any sword`, changing this currently would be a breaking change if anyone unintentionally abused this oversight. Additionally would prevent making random kits like below where each player was given same enchants just a different sword
```
set {_item} to any sword named "" with lore ""
set {_player1}'s tool to {_item}
set {_player2}'s tool to {_item} 
# this will no longer cause player 1 to have stone and player 2 to have netherite they will now be the same.
```

I would label this issue up for debates and wait for more team members to respond or acknowledge the issue, before going forwards with this change. Areas this type of issue might appear are recipes, an example of this can be within Skript 2.8 recipes.

*I haven't actively tested this code as I refuse to download this build, but it should follow same issue if anyone wants to test it*
```
on script load:
  set {_item} to any sword
  create a crafting recipe:
    key: "i_hate_your_keys"
    result: {_item}
    ingredients:
      dirt, dirt and dirt
      {_item}, {_item} and {_item}
      dirt, dirt and dirt
```

I have however tested with basically the exact same code using skbee's wip stuff, which uses nearly the exact same internals as above just without custom effect parsing.
```
on load:
	register shaped recipe with id "testing" for diamond:
		shape: "a", "b", "c"
		ingredients:
			set {_item} to any sword
			set ingredient of "a" to {_item}
			set ingredient of "b" to {_item}
			set ingredient of "c" to {_item}
```
Image below shows how skript currently treats `{_item}` by randomly choosing the item (y) in the `set ingredient x to y` part
expected result was for "a", "b" and "c" to all be the same itemtype instead was given 3 different swords
![image](https://github.com/SkriptLang/Skript/assets/81952476/bc98f164-5f47-41c6-9908-19b0ffd91f12)

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this suggestion.

**Comment by AyhamAl-Ali** [comment]
`any sword` is not an expression, it's an alias, take it as like a Literal, it parses (chooses the item) at runtime, so IMO it's expected, if you want to give random items you would need to use `random element out of {items::*}` or  use `any sword` in your ingredients

**Comment by Fusezion** [comment]
The intention is to not give random when the variable is used repeatedly. I don't recall saying it's an expression sorry if some miss communication went around there. If it was to choose the item at run time it would set the variable to said item at runtime and not change it when the variable is called again. Unless I'm miss understanding the usage of runtime parsing. This style should be treated the same as `set {_var} to slot 0 of player` where we change it to an itemstack at least that's my opinion

As for `random element out of ...` that doesn't solve it either
![image](https://github.com/SkriptLang/Skript/assets/81952476/7df1e800-af02-454a-b4ad-9f45832a592b)
![image](https://github.com/SkriptLang/Skript/assets/81952476/dd700a44-01c3-4e95-a632-784e090d7f81)

**Comment by AyhamAl-Ali** [comment]
I mis-understood you, I though the current state of  the code you provided first doesn't give random each time which is correct. IMO `any sword` or any alias that use `any` should choose the item when it's called such as when setting a var to `any sword` (itemstack not itemtype)

**Comment by TheLimeGlass** [comment]
I do not agree with changing item types to always be saved as ItemTypes. There are Literal cases where it can be used, and also many comparisons.

The randomization currently there is good. We recently had an issue where any sword was not return a random element, and that was fixed. So clearly people want any sword to be true randomization and not pre-fixed, because what's the point of that. any sword should be random, and if you want an exact item, you use the full name like `a diamond sword`

**Comment by AyhamAl-Ali** [comment]
The main issue here Lime is that `set {_item} to any sword` should not stay as `any sword` but rather should be chosen and saved as ItemStack so when someone uses `{_item}` again it will be a single item not randomized on every variable use like as if we called `any sword` directly, because that's the expected and what should happen otherwise users can use `any sword` directly without saving to a variable.

If solving this would cause more issue then maybe we can add some expression to achieve this case specifically

**Comment by Fusezion** [comment]
If anyone's able to get pickle to comment on this as it's normally just Pickle and Lime that I see mess with itemtypes and aliases, I do want his opinion for this before any change is made to this.

**Comment by Fusezion** [comment]
So no further discussion?

**Comment by sovdeeth** [comment]
Category aliases are being sunset. This isn't applicable anymore
